plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/internal/crc/server/ui/PullSecretValidation.java

```diff
@@ -0,0 +1,126 @@
+/******************************************************************************* 
+ * Copyright (c) 2020 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.crc.server.ui;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.osgi.util.NLS;
+import org.jboss.tools.openshift.internal.common.ui.job.UIUpdatingJob;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class PullSecretValidation {
+
+	private static final int MAX_PULLSECRET_FILE_SIZE = 20 * 1024; // 20k
+
+	private final AtomicReference<String> error = new AtomicReference<>();
+	private final AtomicReference<String> file = new AtomicReference<>();
```

Comment:
for better maintenance, maybe rename it as it collide in several places ( constructor, ... )

Reply:
Can you please explain where you see a collision?